### PR TITLE
Catch all content type parser

### PR DESCRIPTION
**Pull request by delvedor**
As titled, see also https://github.com/fastify/fastify/issues/297.

### CODE REVIEW

File: docs/ContentTypeParser.md [hunk 1]
```diff
@@ -16,3 +16,17 @@ fastify.addContentTypeParser('application/jsoff', function (req, done) {
...
+```js
+fastify.addContentTypeParser('*', function (req, done) {
+  var data = ''
+  req.on('data', chunk => { data += chunk })
+  req.on('end', () => {
+    done(data)
+  })
+})
+```
+In this way all the request that haven't a corresponding content type parser will be handled by the above function.
```
Comment (jsumners) [inline docs/ContentTypeParser.md]: > In this way all of the requests that do not have a corresponding content type parser will be handled by the specified function.

^ grammar cleanup

File: lib/ContentTypeParser.js [hunk 2]
```diff
@@ -6,11 +6,20 @@ function ContentTypeParser () {
 }
 
 ContentTypeParser.prototype.add = function (contentType, fn) {
+  if (typeof contentType !== 'string') throw new Error('The content type should be a string')
+  if (contentType.length === 0) throw new Error('The content type cannot be an empty string')
+  if (typeof fn !== 'function') throw new Error('The content type handler should be a function')
```
Comment (allevo) [inline lib/ContentTypeParser.js]:
```js
assert.ok(typeof contentType !== 'string', 'The content type should be a string')
assert.equal(contentType.length, 0, 'The content type cannot be an empty string')
assert.ok(typeof fn !== 'function', 'The content type handler should be a function')
```

File: test/custom-parser.test.js [hunk 3]
```diff
@@ -251,3 +251,142 @@ test('contentTypeParser shouldn\'t support request with undefined "Content-Type"
...
+    req.on('data', chunk => { data += chunk })
+    req.on('end', () => {
+      done(data)
+    })
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+
+    request({
```
Comment (allevo) [inline test/custom-parser.test.js]: Why not`inject`?

File: lib/ContentTypeParser.js (hunk 2 above)
Comment (delvedor) [reply lib/ContentTypeParser.js]: It's the same thing, I don't think is needed.

File: test/custom-parser.test.js (hunk 3 above)
Comment (delvedor) [reply test/custom-parser.test.js]: Because this test has been created before the api `inject` :joy:

File: lib/ContentTypeParser.js (hunk 2 above)
Comment (jsumners) [reply lib/ContentTypeParser.js]: Yeah, there is no need to add an extra function invocation for no reason.

Comment (delvedor) [reply lib/ContentTypeParser.js]: @jsumners this is not "hot" code so an extra function invocation is not a problem. In any case use `assert` does not give any benefit.
If we want to use it, we should update the entire framework to be more consistent.

Comment (allevo) [reply lib/ContentTypeParser.js]: Yeah. Maybe we'll update the entire code 😨

**Pull request merged**